packages/core/src/external/carequality/ihe-gateway-v2/saml/saml-client.ts

```diff
@@ -102,5 +105,128 @@ export async function sendSignedXmlMtom({
     },
     httpsAgent: agent,
   });
+
+  try {
+    const mtomContentType = parseMtomContentType(response.headers["content-type"]);
+    const binaryData: Buffer = Buffer.isBuffer(response.data)
+      ? response.data
+      : Buffer.from(response.data);
+    const mtomParts = await parseMTOMResp(binaryData, mtomContentType.boundary);
+    const soapData: Buffer = mtomParts.parts[0]?.body || Buffer.from("");
+    const fileData: Buffer = mtomParts.parts[1]?.body || Buffer.from("");
+    const fileType = detectFileType(fileData);
+    const fileType2 = detectFileType(soapData);
+    console.log("File type:", fileType);
+    console.log("File type:", fileType2);
```

Comment:
the file types are right here but the fileData when written to a file or uploaded to s3 is corrupted